app/models/customer.rb

```diff
@@ -35,6 +35,7 @@ class Customer < ApplicationRecord
                     uniqueness: { scope: :enterprise_id, message: I18n.t('validation_msg_is_associated_with_an_exising_customer') }
 
   scope :of, ->(enterprise) { where(enterprise_id: enterprise) }
+  scope :managed_by, ->(user) { user&.persisted? ? where(user: user).or(of(Enterprise.managed_by(user))) : none }
```

Comment:
I'm still not sure about `where(user: user)`. This would allow a normal shopper (with API access) to edit their customer entry, right? That includes the enterprise id.

This is a pre-existing issue, so maybe out of scope here but pretty important. Two issues:

1. Do we let buyers change their customer records via the API or only enterprise managers?
2. We shouldn't allow updating the enterprise id. Updating the email can be dangerous as well but that risk is up to the enterprise manager, I think.

Reply:
maybe @lin-d-hop can help?

Reply:
This endpoint is intended to mimic admin/customers. Not /account.

A user should not be able to edit their personal customer details unless they are a manager of the hub they are editing themself as a customer of.

Agree enterprise-id should not be editable.
S2 bug.